### ISSUE TITLE
Pull/Push review (mapping control data elements)

### ISSUE THREAD

**Issue by xurxodev** [ifoche/malariapp#2427]

**Comment by xurxodev** [comment]
@ifoche I have reviewed the push and pull and the code for CDE about scoring is valid, we have not removed this code when we have changed to competencies.

During the testing creating surveys with KE HNQIS Family Planning Counseling I have found conflict in the push and then the scoring is not saved in the server. The pull doesn't assign a score for these events because any value exists in the server for these events after the conflict and in the list, the scores appear as  -.

The problem is that PSI is mixing the concept of mandatory and critical questions, for this program there are critical questions as children of different parent questions. The app allow complete a survey with all current critical questions answered but is possible under a parent question answered as no that exists more critical and mandatory questions, the app allows complete but dhis2 response conflict because there are mandatory data element without a value in the event.

The client report us a similar problem in this issue https://github.com/EyeSeeTea/malariapp/issues/2419
